### PR TITLE
[skip ci] Improve docs on getting started

### PR DESCRIPTION
**Pull request by yorinasub17**
This improves the docs on the getting started experience with terratest. We have had a few inquiries and sources of confusion in terms of how one should get started with terratest. This attempts to address that by having detailed step-by-step guide on how one should start using terratest with their modules.

Reference: https://github.com/gruntwork-io/terratest/issues/316

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,21 +62,43 @@ validateServerIsWorking(t, terraformOptions)
...
 
-Prerequisite: install [Go](https://golang.org/).
+### Install requirements
 
-To add Terratest to your projects, we recommend using a Go dependency manager such as
-[dep](https://github.com/golang/dep) to add the packages you wish to use (see [package by package overview](#package-by-package-overview) for the list). For example, to add the `terraform` package:
+Terratest uses the Go testing framework. To use terratest, you need to install:
+
+- [Go](https://golang.org/) (requires version >=1.10)
+- [dep](https://github.com/golang/dep) (requires version >=0.5.1)
```
Comment (yorinasub17) [inline README.md]: With the `kubernetes` repos, it is no longer a good idea to use `go get`.

File: README.md [hunk 2]
```diff
@@ -62,21 +62,43 @@ validateServerIsWorking(t, terraformOptions)
...
+
+
+### Setting up your project
+
+1. Golang requires go source files to be available in the `GOPATH`. By default this is typically `$HOME/go/src`. Create a new blank folder in the `GOPATH` to hold your terraform module and terratest code. For example, if you were developing a VPC module named `terraform-aws-vpc`, create the folder `$HOME/go/src/terraform-aws-vpc` to hold your module code.
```
Comment (yorinasub17) [inline README.md]: Once we have go modules, life would be a lot better. Esp with all the k8s stuff since k8s now officially supports go modules.

File: README.md [hunk 3]
```diff
@@ -62,21 +62,43 @@ validateServerIsWorking(t, terraformOptions)
...
+  version = "0.17.4"
 ```
 
-Alternatively, you can use `go get`:
+Now you should be able to run the example test. To run the test:
 
-```bash
-go get github.com/gruntwork-io/terratest/modules/terraform
+1. Change your working directory to the `test` folder.
+1. Run `dep ensure`
```
Comment (brikis98) [inline README.md]: Perhaps add a note that this downloads dependencies, so you only need to run it once?

File: README.md [hunk 4]
```diff
@@ -62,21 +62,43 @@ validateServerIsWorking(t, terraformOptions)
...
+1. Run `go test -v .`
```
Comment (brikis98) [inline README.md]: Then rephrase this to "Each time you want to run tests, use".

File: README.md [hunk 5]
```diff
@@ -62,21 +62,43 @@ validateServerIsWorking(t, terraformOptions)
...
+
+Note that `go` has a default test timeout of 10 minutes. With infrastructure testing, your tests will surpass the 10 minutes very easily. To extend the timeout, you can pass in the `-timeout` option, which takes a `go` duration string (e.g `10m` for 10 minutes or `1h` for 1 hour). For example, to run the tests with a 90 minute timeout:
```
Comment (brikis98) [inline README.md]: 1. Make `-timeout` part of the default command above. It should *always* be there for infrastructure testing.
1. Expand this explanation to explain that if you hit the timeout, Go kills the test, preventing it from finishing, and preventing the cleanup code from running.

**Comment by yorinasub17** [comment]
Addressed all comments in fd73724

**Comment by yorinasub17** [comment]
Going to go ahead and merge this. Thanks for the reviews!
